### PR TITLE
WIP: outdoor combo landuse pois boundary

### PR DESCRIPTION
**Pull request by nvkelso**
Connects with #663 for adding outdoor combo landuse, pois, and boundaries.

* [filter] add outdoor combo landuse, poi, boundary features
* [sort_key] adjust sort_key for new polygon and line outdoor features
* [docs] update layer reference for new landuse, pois, and boundary featuers
* [test] add tests for combo outdoor landuse, pois, boundary features
* [migration] add migrations for new outdoor combo landuse, pois, boundary

### CODE REVIEW

**Comment by nvkelso** [comment]
Some tests are inexplicably not passing. They're commented out for now, will return to this WIP PR.

File: docs/layers.md [hunk 1]
```diff
@@ -672,6 +693,7 @@ Features from OpenStreetMap which are tagged `disused=*` for any other value tha
 * `photographic_laboratory`
 * `physician`
 * `picnic_site`
+* `picnic_site`
```
Comment (nvkelso) [inline docs/layers.md]: This is a dup.

File: test/663-combo-outdoor-landuse-pois.py [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+#https://www.openstreetmap.org/node/4050178586
+# missing
+# Camp Goodtimes
+# assert_has_feature(
+#     15, 5225, 11211, 'pois',
+#     { 'kind': 'summer_camp', 'min_zoom': 15 })    
+
+
+
+#https://www.openstreetmap.org/node/3838356961
```
Comment (nvkelso) [inline test/663-combo-outdoor-landuse-pois.py]: No battlefields are exporting.

File: test/663-combo-outdoor-landuse-pois.py [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+
+
+#https://www.openstreetmap.org/node/3655879348
+# Camp Ahmek
+assert_has_feature(
+    15, 9219, 11714, 'pois',
+    { 'kind': 'summer_camp', 'min_zoom': 15 })
+    
+#https://www.openstreetmap.org/node/4050178586
+# missing
```
Comment (nvkelso) [inline test/663-combo-outdoor-landuse-pois.py]: I don't have this in my local DB, need to import from OSM.org

File: test/663-combo-outdoor-landuse-pois.py [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+#https://www.openstreetmap.org/node/3297410094
+# Why is this missing?
+# Golden Gate Park, SF
+assert_has_feature(
+    16, 10472, 25332, 'pois',
+    { 'kind': 'picnic_site', 'min_zoom': 16 })
+
+#https://www.openstreetmap.org/way/400701941
+# missing landuse feature? needs import?
```
Comment (nvkelso) [inline test/663-combo-outdoor-landuse-pois.py]: I don't have this in my local DB, need to import from OSM.org

File: test/663-combo-outdoor-landuse-pois.py [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+    { 'kind': 'monument', 'min_zoom': 16.2397 })
+
+#https://www.openstreetmap.org/way/248460669
+# building, Jefferson Monument
+assert_has_feature(
+    15, 9371, 12537, 'pois',
+    { 'kind': 'monument', 'min_zoom': 15 })
+
+#https://www.openstreetmap.org/way/66418767
+# huh? this in in the DB, but missing from tiles
```
Comment (nvkelso) [inline test/663-combo-outdoor-landuse-pois.py]: Perhaps the building part confuses things?

File: test/663-combo-outdoor-landuse-pois.py [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+assert_has_feature(
+    12, 685, 1580, 'landuse',
+    { 'kind': 'dam', 'sort_key': 221 })
+
+assert_has_feature(
+    12, 685, 1580, 'pois',
+    { 'kind': 'dam', 'min_zoom': 12 })
+
+#https://www.openstreetmap.org/way/62201624
+# huh, missing in tile, is in DB
```
Comment (nvkelso) [inline test/663-combo-outdoor-landuse-pois.py]: boundary layer + line features from OSM don't seem to be configured right yet.

**Comment by nvkelso** [comment]
@zerebubuth Anything obviously funky for the missing `battlefield` poi and `dam` boundary line?

File: test/663-combo-outdoor-landuse-pois.py (hunk 2 above)
Comment (zerebubuth) [inline test/663-combo-outdoor-landuse-pois.py]: I'm seeing the Battle of Blackburn's Ford battlefield POI OK:

```json
    {
      "geometry": {
        "type": "Point",
        "coordinates": [
          -77.44954723,
          38.80331938
        ]
      },
      "type": "Feature",
      "properties": {
        "kind": "battlefield",
        "min_zoom": 17,
        "id": 3838356961,
        "name": "Battle of Blackburn's Ford (1861)"
      }
    }
```

I'm missing the Kernstown POI from my local DB, so couldn't test that one.

The Antietam battlefield is getting a `min_zoom` of 10.4417 in my local DB, which means it only kicks in at zoom 11 (in tile 11/581/779):

```json
    {
      "geometry": {
        "type": "Point",
        "coordinates": [
          -77.7362,
          39.4678
        ]
      },
      "type": "Feature",
      "properties": {
        "id": 231393152,
        "kind": "battlefield",
        "min_zoom": 10.4417,
        "name": "Antietam National Battlefield",
        "area": 12969200
      }
    },
```

The polygon is also present in `landuse` (again, at zoom 11), but as a `national_park`:

```json
      "type": "Feature",
      "properties": {
        "kind": "national_park",
        "source": "openstreetmap.org",
        "sort_key": 18,
        "area": 12969200,
        "mz_is_building": false,
        "min_zoom": 5.44171,
        "id": 231393152
      }
```

The filter for `boundary=national_park` => `kind: national_park` is the highest one in the YAML file, so take priority.

**Comment by nvkelso** [comment]
Seriously, I don't know what possessed me to make such a complicated issue that deserved 60 tests. They all pass now. merging.
